### PR TITLE
Fix a few typos in function comments

### PR DESCRIPTION
**Pull request by owen-mc-diffblue**
The only non-trivial change is object_descriptor_offsett ->
object_descriptor_exprt

### CODE REVIEW

**Review by smowton — APPROVED** [review]

File: src/pointer-analysis/value_set.h [hunk 1]
```diff
@@ -163,7 +163,7 @@ class value_sett
   };
 
   /// Converts an `object_map_dt` entry `object_number -> offset` into an
-  /// `object_descriptor_offsett` with
+  /// `object_descriptor_exprt` with
   /// `.object() == object_numbering.at(object_number)` and
```
Comment (chrisr-diffblue) [inline src/pointer-analysis/value_set.h:167]: If you are fixing up this comment, is there any milage in converting it to proper doxygen format (with `\param` notation) at the same time? Won't hold it against you if you don't want to though :-)

**Review by chrisr-diffblue — APPROVED** [review]
Looks good to me. Minor comment, but happy for you to ignore as this is still a cleanup as it is.

File: src/pointer-analysis/value_set.h (hunk 1 above)
Comment (smowton) [reply src/pointer-analysis/value_set.h:167]: I deliberately skipped `\param it: thing being converted; \return converted thing` cos I didn't think it added anything, but don't object if it helps the formatted docs? (I've never seen our actual Doxy output, so I'm flying blind a bit when writing this format)

Comment (chrisr-diffblue) [reply src/pointer-analysis/value_set.h:167]: I've not actually seen them either, so equally blind :-) But I think it should help with the formatting and hyperlinking. But I really don't have a strong opinion here either.

**Pull request merged**